solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -155,6 +210,68 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         return amount;
     }
 
+    /**
+      * @dev returns the recent average rate of 1 `_token` in the other reserve token units
+      * note that the rate can only be queried for reserves in a standard pool
+      *
+      * @param _token   token to get the rate for
+      * @return recent average rate between the reserves (numerator)
+      * @return recent average rate between the reserves (denominator)
+    */
+    function recentAverageRate(IERC20Token _token) external view returns (uint256, uint256) {
+        // if the pool isn't standard, return 0, 0
+        if (!isStandardPool) {
+            return (0, 0);
+        }
+
+        // get the recent average rate of reserve 0
+        Fraction memory rate = recentAverageRate();
+        if (_token == reserveTokens[0]) {
+            return (rate.n, rate.d);
+        }
+
+        return (rate.d, rate.n);
+    }
+
+    /**
+      * @dev returns the recent average rate of 1 reserve token 0 in reserve token 1 units
+      *
+      * @return recent average rate between the reserves
+    */
+    function recentAverageRate() internal view returns (Fraction memory) {
+        // get the elapsed time since the previous average rate was calculated
+        uint256 timeElapsed = time() - prevAverageRateUpdateTime;

```

Comment:
why?

Reply:
Because in `time() - prevAverageRateUpdateTime`, you are assuming that `time() >= prevAverageRateUpdateTime`.

Reply:
it's good enough since we can also introduce a function that updates `prevAverageRateUpdateTime` in the tests if needed.
I actually did and then removed it - it's better to increase the time than move it back and forth.

Reply:
So whatever function we put in the tester contract should be named `incTime` instead of `setTime`.

Reply:
it's all under the test responsibility (including the custom setters) so I don't see the significance in doing that, better keep it simple in the tests.